tests/common/txn_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/tests/v3/framework"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+type txnReq struct {
+	compare  []string
+	ifSucess []string
+	ifFail   []string
+	results  []string
+}
+
+func testTxn(t *testing.T, cfgs []config.NamedClusterConfig, setup func(framework.Client), reqs []txnReq) {
+	testRunner.BeforeTest(t)
+	for _, cfg := range cfgs {
+		t.Run(cfg.Name, func(t *testing.T) {
+			clus := testRunner.NewCluster(t, cfg.Config)
+			defer clus.Close()
+			cc := clus.Client()
+			testutils.ExecuteWithTimeout(t, 10*time.Second, func() {
+				setup(cc)
+				for _, req := range reqs {
+					resp, err := cc.Txn(req.compare, req.ifSucess, req.ifFail, config.TxnOptions{
+						Interactive: true,
+					})
+					if err != nil {
+						t.Errorf("Txn returned error: %s", err)
+					}
+					assert.Equal(t, req.results, getRespValues(resp))
+				}
+			})
+		})
+	}
+}
+
+func TestTxnSucc(t *testing.T) {
+	tcs := []config.NamedClusterConfig{
+		{
+			Name:   "NoTLS",
+			Config: config.ClusterConfig{ClusterSize: 1},
+		},
+		{
+			Name:   "PeerTLS",
+			Config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.ManualTLS},
+		},
+		{
+			Name:   "PeerAutoTLS",
+			Config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.AutoTLS},
+		},
+		{
+			Name:   "ClientTLS",
+			Config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.ManualTLS},
+		},
+		{
+			Name:   "ClientAutoTLS",
+			Config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.AutoTLS},
+		},
```

Comment:
@serathius this test config seems common in all test cases in `tests/common` , can we extract it out in a separate variable?

Reply:
yea I think this is a good idea. I think can think about doing it in separate PR.